src/ORM/DataList.php

```diff
@@ -591,20 +581,13 @@ private function getFilterAnySubquery(DataQuery $query, array $whereArguments):
     /**
      * Note that, in the current implementation, the filtered list will be an ArrayList, but this may change in a
      * future implementation.
-     * @see Filterable::filterByCallback()
+     * @see SS_List::filterByCallback()
      *
      * @example $list = $list->filterByCallback(function($item, $list) { return $item->Age == 9; })
-     * @param callable $callback
      * @return ArrayList<T>
      */
-    public function filterByCallback($callback)
+    public function filterByCallback(callable $callback): ArrayList
```

Comment:
```suggestion
    public function filterByCallback(callable $callback): SS_List
```
Lets allow ourselves to change that in the future. From an API perspective devs only need to know it has the `SSList` set of methods.
This lets us potentially return a `DataList` in the future for example, if we find a way to do that.

Reply:
Updated

Also updated the same return type on ListDecorator::filterByCallback() and EagerLoadedList::filterByCallback()